python/HeaderService/camera_coords.py

```diff
@@ -138,11 +158,11 @@ def __init__(self, vendor,
                  overv=None,
                  xpixelscale=0.105,
                  ypixelscale=0.105,
+                 crpix1=None,
+                 crpix2=None,
                  ):
         self.vendor = vendor
```

Comment:
I think that since you are doing:
```python
if self.vendor == "ITL":
    ...
elif self.vendor == "E2V"
   ...
```
without having an `else` I think it would be a good idea to validate `vendor` here to ensure that you know things have gone wrong early.

Reply:
The check for the allowed `vendor` is done when the function `HDR.load_templates()` is called. This happens at startup and when a new image is triggered. At the moment the type of vendor is set up by the HeaderService config file at startup (`config.vendor`). However, after the last `ts_xml` update that information is now available from `ATCamera` via the event `ATCamera.wrebSettingsApplied.ccdManufacturer`. I think we should change the current implementation and grab the info from the event instead, as this would be the need for LSSTCam too. I would like to do this on a separate ticket as it doesn't relate to the WCS.